File: EmsBMSReports/dlgSelectColumns.cs
using System;
using static MactusReportLib.MactusReportLib;
using Microsoft.VisualBasic.CompilerServices;

namespace EmsBMSReports
{
    public partial class dlgSelectColumns
    {
        public dlgSelectColumns()
        {
            InitializeComponent();
        }
        private void dlgSelectColumns_Load(object sender, EventArgs e)
        {
            foreach (var col in g_oColList)
            {
                oColumns.Items.Add(col.m_sColTitle);
                if (col.m_bshowAlarmCol)
                {
                    oColumns.SetItemChecked(oColumns.Items.Count - 1, true);
                }
            }
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            Close();
        }

        private void oColumns_SelectedIndexChanged(object sender, EventArgs e)
        {
            foreach (var col in g_oColList)
                col.m_bshowAlarmCol = false;
            foreach (var col in oColumns.CheckedIndices)
                g_oColList[Conversions.ToInteger(col)].m_bshowAlarmCol = true;
        }
    }
}

File: EmsBMSReports/dlgSelectColumns.vb
Imports MactusReportLib.MactusReportLib
Public Class dlgSelectColumns
    Private Sub dlgSelectColumns_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        For Each col In g_oColList
            oColumns.Items.Add(col.m_sColTitle)
            If col.m_bshowAlarmCol Then
                oColumns.SetItemChecked(oColumns.Items.Count - 1, True)
            End If
        Next
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Close()
    End Sub

    Private Sub oColumns_SelectedIndexChanged(sender As Object, e As EventArgs) Handles oColumns.SelectedIndexChanged
        For Each col In g_oColList
            col.m_bshowAlarmCol = False
        Next
        For Each col In oColumns.CheckedIndices
            g_oColList(col).m_bshowAlarmCol = True
        Next
    End Sub
End Class